<a href="https://colab.research.google.com/github/pavi-ninjaac/Kaggle/blob/main/Digit_Recognition.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [10]:
# the aim is to predict the numbers from 0-9
#import libraries

import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

#Sklearn packages for data preparation
from sklearn.model_selection import train_test_split
from sklearn.metrics import confusion_matrix

In [3]:
from google.colab import drive
drive.mount('/content/gdrive')

Mounted at /content/gdrive


In [5]:
with open('/content/gdrive/MyDrive/digit-recognizer/train.csv') as f:
  line = f.readlines()

row = np.array(line)
no_of_rows = row.size
print("Number of data is-->",no_of_rows)

Number of data is--> 42001


The Size is 42001, 42000 is data firt line is the heading we can leave this

In [9]:
X,y = [],[]

#------------------------------------------
#Transfer the data into train and test dataset
for index in range(1,no_of_rows):
  label,*img = row[index].split(',')
  img = np.array(img)
  y.append(label)
  X.append(img)
print("Lenth of the Dataset",len(y))

Lenth of the Dataset 42000


In [13]:
# covert list to a pandas DataFrame 
X = pd.DataFrame(X,index=None)
X.head(3)

0   1   2   3   4   5   6   7   8    ... 775 776 777 778 779 780 781 782  783
0   0   0   0   0   0   0   0   0   0  ...   0   0   0   0   0   0   0   0  0\n
1   0   0   0   0   0   0   0   0   0  ...   0   0   0   0   0   0   0   0  0\n
2   0   0   0   0   0   0   0   0   0  ...   0   0   0   0   0   0   0   0  0\n

[3 rows x 784 columns]

In [ ]:
#split it as train and test datasets
#-------------------------------------------
X_train,X_test,y_train,y_test = train_test_split(X,y,test_size=0.25,random_state = 42)